optuna/structs.py

```diff
@@ -96,7 +93,81 @@ class FrozenTrial(
             :class:`~optuna.study.Study.study_id` to identify a :class:`~optuna.trial.Trial`.
     """
 
-    internal_fields = ['distributions', 'trial_id']
+    def __init__(
+        self,
+        number,  # type: int
+        state,  # type: TrialState
+        value,  # type: Optional[float]
+        datetime_start,  # type: Optional[datetime]
+        datetime_complete,  # type: Optional[datetime]
+        params,  # type: Dict[str, Any]
+        distributions,  # type: Dict[str, BaseDistribution]
+        user_attrs,  # type: Dict[str, Any]
+        system_attrs,  # type: Dict[str, Any]
+        intermediate_values,  # type: Dict[int, float]
+        trial_id,  # type: int
+    ):
+        self.number = number
+        self.state = state
+        self.value = value
+        self.datetime_start = datetime_start
+        self.datetime_complete = datetime_complete
+        self.params = params
+        self.distributions = distributions
+        self.user_attrs = user_attrs
+        self.system_attrs = system_attrs
+        self.intermediate_values = intermediate_values
+        self._trial_id = trial_id
+
+    internal_fields = ['distributions', '_trial_id']
+    _fields = [
```

Comment:
It seems a bit cumbersome to maintain the field list manually. How about using a meta-programming feature (e.g., `__dict__`) of Python to generate this list automatically?

Reply:
Thank you for the suggestion. I checked the `__dict__`s of the returned frozen trials from the storage but as they were empty, I'll investigate.